lib/ansible/modules/system/systemd.py

```diff
@@ -457,6 +487,21 @@ def main():
                         (rc, out, err) = module.run_command("%s %s '%s'" % (systemctl, action, unit))
                         if rc != 0:
                             module.fail_json(msg="Unable to %s service %s: %s" % (action, unit, err))
+
+                        if module.params['detect_early_failure']:
```

Comment:
does this make sense with a stop? in most cases 'reload' is just sending a signal to the daemon ... not sure it applies to that action either.